### PR TITLE
feat: add sort function to age

### PR DESCRIPTION
**Pull request by sae-github**
# [Lesson22](https://github.com/sae-github/handsonFrontend/blob/master/work/markup/1.md#22)

> こちらのようなテーブルを画面遷移してから3秒後に解決されるPromiseが返すオブジェクトを元に作り、 idがソートできる機能を作ってください
> <a href="https://gyazo.com/e1f8e3995c83547a0d47100c29268d1f">Image from Gyazo</a>
> - ソートは通常時はidが適当でもよく
> - ソートが昇順の場合は上矢印がアクティブ、下矢印がdisabled、1,2,3,4,5の順番で表示され、降順の場合はその逆、
通常時の矢印クリック(クリッカブル領域は2つの矢印です。上下別々のクリッカブル領域でではなく)を押すと画像のように変化します
> - 同じことを年齢でもやってください。

### [ comment ]
- Previously, elements were retrieved using HTML tags, but this has been changed to using js- to retrieve elements by assigning class names.
- When the sort button is clicked, the status of the sort function will be changed to default except for the clicked element.

 I would be happy to receive your review :)

### CODE REVIEW

File: lesson22/index.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  return table;
+};
+
+const createTableHead = (items) => {
+  const thead = document.createElement("thead");
+  const tr = document.createElement("tr");
+  Object.values(items).forEach((item) => {
+    const th = createElementWithClassName("th", "js-th");
+    th.textContent = item;
+    (item === "ID" || item === "年齢") && th.appendChild(createSortButton());
```
Comment (ghost) [inline lesson22/index.js]: IMO
Is it possible to display text dynamically instead of a fixed string in order to be able to support additional features?

Comment (sae-github) [reply lesson22/index.js]: Thanks for the review :)

Changed the way `th` is generated to assign an id to each element.
In `addSortButton`, the element to which you want to add a `sortButton` is obtained by its id, and `createSortButton` is executed.
We also changed the process for sorting to be executed in `initSortButton`.

I have made some modifications, but I am not sure if this meets the requirements of being dynamic.
Please check.

e3023ee

**Review by ghost — APPROVED** [review]

File: lesson22/index.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      );
+    case "desc":
+      return [...defaultRows].sort(
+        (a, b) => b.children[index].textContent - a.children[index].textContent
+      );
+    default:
+      throw new Error(`${status} is not provided.`);
+  }
+};
+
```
Comment (csakyo) [inline lesson22/index.js:196]: I've never created a sort function before, but I didn't know there was such a way! I thought your code was very clear.

**Review by csakyo — APPROVED** [review]
I thought the variable and function names were very clear and it was easy to imagine what was being done.

File: lesson22/index.js (hunk 2 above)
Comment (sae-github) [reply lesson22/index.js:196]: Thanks for the review.

One of my goals was to write code that was easy for others to understand when they saw it. Your words give me confidence.☺️

**Pull request merged**